crates/executor/src/estimate.rs

```diff
@@ -39,7 +39,21 @@ pub fn estimate(
             blockifier::transaction::objects::FeeType::Eth => PriceUnit::Wei,
         };
 
-        let tx_info = transaction
+        let minimal_l1_gas_amount_vector = match &transaction {
+            Transaction::AccountTransaction(account_transaction) => Some(
+                blockifier::fee::gas_usage::estimate_minimal_gas_vector(
+                    &block_context,
+                    &account_transaction,
+                )
+                .map_err(|e| TransactionExecutionError::new(transaction_idx, e.into()))?,
+            ),
+            Transaction::L1HandlerTransaction(_) => None,
```

Comment:
Why are L1 Handlers excluded from this requirement?

Reply:
L1 handlers are not AccountTransactions. `estimate_minimal_gas_vector` works only for account transactions and as far as I can understand the fee bounds check is also only done for those.